mulighetsrommet-api/src/main/kotlin/no/nav/mulighetsrommet/api/refusjon/ArrangorflateRoutes.kt

```diff
@@ -110,26 +100,48 @@ fun Route.arrangorflateRoutes() {
 
             post("/godkjenn-refusjon") {
                 val id = call.parameters.getOrFail<UUID>("id")
-
                 val krav = refusjonskrav.get(id)
                     ?: throw NotFoundException("Fant ikke refusjonskrav med id=$id")
-
                 requireTilgangHosArrangor(krav.arrangor.organisasjonsnummer)
 
-                val request = call.receive<GodkjennRefusjonskravAft>()
+                val request = call.receive<GodkjennRefusjonskrav>()
+                validerGodkjennRefusjonskrav(request, krav)
+                    .onLeft { return@post call.respondWithStatusResponse(BadRequest(errors = it).left()) }
+
+                db.transactionSuspend { tx ->
+                    refusjonskrav.setGodkjentAvArrangor(id, LocalDateTime.now(), tx)
+                    refusjonskrav.setBetalingsInformasjon(
+                        id,
+                        request.betalingsinformasjon.kontonummer,
+                        request.betalingsinformasjon.kid,
+                        tx,
+                    )
+
+                    val pdf = run {
+                        val tilsagn = tilsagnService.getArrangorflateTilsagnTilRefusjon(
+                            gjennomforingId = krav.gjennomforing.id,
+                            periode = krav.beregning.input.periode,
+                        )
+                        val refusjonsKravAft: RefusjonKravAft = toRefusjonskrav(pdl, deltakerRepository, krav)
+                        Pdfgen.refusjonJournalpost(refusjonsKravAft, tilsagn)
+                    }
 
-                val result = validerGodkjennRefusjonskrav(request, krav)
-                    .mapLeft { BadRequest(errors = it) }
-                    .map { (betalingsinformasjon) ->
-                        refusjonskrav.setGodkjentAvArrangor(id, LocalDateTime.now())
-                        refusjonskrav.setBetalingsInformasjon(
-                            id,
-                            betalingsinformasjon.kontonummer,
-                            betalingsinformasjon.kid,
+                    val result = dokarkClient.opprettJournalpost(
+                        refusjonskravJournalpost(pdf, krav.id, krav.arrangor.organisasjonsnummer),
+                        AccessType.M2M,
+                    )
+                    when (result) {
+                        is DokarkResult.Error -> throw StatusException(
+                            HttpStatusCode.InternalServerError,
+                            "Feilet ved opprettelse av journalpost",
                         )
+                        is DokarkResult.Success -> {
+                            refusjonskrav.setJournalpostId(id, result.journalpostId, tx)
+                        }
                     }
```

Comment:
Dette kunne muligens vært en kandidat for en `JournalforRefusjonskravTask`. Altså en one-time task som tar inn refusjonskrav-id som input.

Noen fordeler:
- Mindre jobb som må gjøres når man godkjenner et refusjonskrav, alt gjøres i samme databasetransaksjon
- Vi skal antagelig kommunisere med oebs her etter hvert også, da kan vi ende i en rar tilstand hvis man skal kommunisere med flere tredjepartssystemer som del av den samme transaksjonen. Da gir det kanskje mening å lagre denne oebs-kommunikasjonen som en one-time task også.

Reply:
Jepp enig. Tenkte begynne enkelt, men kanskje skulle gjort noe her med en gang